tests/csaf_2_1/mandatoryTest_6_1_37.js

```diff
@@ -0,0 +1,30 @@
+import assert from 'node:assert/strict'
+import {
+  isValidDate,
+  mandatoryTest_6_1_37,
+} from '../../csaf_2_1/mandatoryTests/mandatoryTest_6_1_37.js'
+
+describe('mandatoryTest_6_1_37', function () {
+  it('only runs on relevant documents', function () {
+    assert.equal(mandatoryTest_6_1_37({ document: 'mydoc' }).isValid, true)
+  })
+
+  describe('isValidDate', function () {
+    const testCases = /** @type {Array<[string, boolean]>} */ ([
+      ['2024-01-01T00:00:00Z', true],
+      ['2024-01-01T00:00:00.000Z', true],
+      ['2024-01-01T00:00:00.0Z', true],
+      ['2024-01-01T00:00:00.11111111Z', true],
+      ['2024-01-01T00:00:00+01:00', true],
+      ['2024-01-01T00:00:00.111111+01:00', true],
```

Comment:
I would like to add a few more things here:
```suggestion
      ['2024-01-01T00:00:00.111111+01:00', true],
      ['2024-02-29T00:00:00.987564+01:00', true],
      ['2015-06-30T10:29:60-13:30', true],
      ['2015-06-30T23:59:60+00:00', true],
      ['2015-07-01T06:59:60+07:00', true],
      ['2016-12-31T00:00:60+23:59', true],
      ['2016-12-31T23:59:60+00:00', true],
      ['2017-01-01T02:59:60+03:00', true],
      ['2017-01-01T02:59:60+04:00', false],
      ['2024-02-30T00:00:00+01:00', false],
      ['2024-04-31T00:00:00+01:00', false],
      ['2024-13-31T00:00:00+01:00', false],

Reply:
Mhm, I'm not sure if just the boolean was flipped but many of these cases fail for me:

## These dates are invalid, but marked as valid in the sample
- `2015-06-30T10:29:60-13:30`
- `2015-06-30T10:29:60-13:30`
- `2015-06-30T23:59:60+00:00`
- `2015-07-01T06:59:60+07:00`
- `2016-12-31T00:00:60+23:59`
- `2016-12-31T23:59:60+00:00`
- `2017-01-01T02:59:60+03:00`
- `2017-01-01T02:59:60+04:00`

## These dates are valid, but marked as invalid in the sample
- `2024-02-30T00:00:00+01:00`
- `2024-04-31T00:00:00+01:00`

Reply:
> Mhm, I'm not sure if just the boolean was flipped but many of these cases fail for me:
> ## These dates are invalid, but marked as valid in the sample
> 
>   * `2015-06-30T10:29:60-13:30`
>   * `2015-06-30T10:29:60-13:30`
>   * `2015-06-30T23:59:60+00:00`
>   * `2015-07-01T06:59:60+07:00`
>   * `2016-12-31T00:00:60+23:59`
>   * `2016-12-31T23:59:60+00:00`
>   * `2017-01-01T02:59:60+03:00`
>   * `2017-01-01T02:59:60+04:00`
> 
> ## These dates are valid, but marked as invalid in the sample
> 
> * `2024-02-30T00:00:00+01:00`
> * `2024-04-31T00:00:00+01:00`

I'm pretty sure the boolean is right. The dates covered in the valid part refer to leap seconds.

For the invalid part: According to my count the February 2024 just had 29 days, not 30 and the April 30, not 31.

Am I missing something?

Reply:
I think, the dates covered in the valid part have `60` seconds which is invalid IMHO. I think in RFC 3339, `10:29:60` is expressed as `10:30:00` e.g.

For the invalid part: `2024-02-30T00:00:00+01:00` is marked with timezone offset `+01:00` which converts to the following `UTC` date: `2024-02-29T23:00:00Z` which is valid, since (as you noted) february had 29 days.

Reply:
> I think, the dates covered in the valid part have `60` seconds which is invalid IMHO.

Sorry - it is not: Please see [RFC 3339 section 5.6](https://www.rfc-editor.org/rfc/rfc3339#section-5.6):
```
time-second     = 2DIGIT  ; 00-58, 00-59, 00-60 based on leap second
```
Those should be date-times of leap seconds (if I didn't miscalculate them).

Reply:
> For the invalid part: `2024-02-30T00:00:00+01:00` is marked with timezone offset `+01:00` which converts to the following `UTC` date: `2024-02-29T23:00:00Z` which is valid, since (as you noted) february had 29 days.

IMHO, that is an incorrect conversion of dates as the February is not allowed to have 30 days in the first place. Please see [RFC 3339 section 5.6](https://www.rfc-editor.org/rfc/rfc3339#section-5.6):

```
date-mday       = 2DIGIT  ; 01-28, 01-29, 01-30, 01-31 based on
                                            ; month/year
```

Reply:
Okay, I updated the `ajv-formats` library and switched to a `date-time` schema check. This now validates the given samples excluding one:

`2016-12-31T00:00:60+23:59`

This is reported as an invalid date.

Reply:
My bad: It must be `2016-12-31T00:00:60-23:59`. Sorry - I forgot to correct that. Is that just an issue here or does that also occur in the OASIS test data.